modules/thermo-cycler/thermo-cycler-arduino/thermo-cycler-arduino.ino

```diff
@@ -696,6 +899,28 @@ void TC4_Handler()
   }
 }
 
+void check_saved_offsets()
+{ // If erroneous value or no offsets saved in EEPROM, then save the defaults to EEPROM
+  if (isnan(eeprom.get_offset_constant(OffsetConst::A)))
```

Comment:
this will work but it makes me feel EXTREMELY bad to do it this way. this is a check on byte values that should really be done _at the level of the byte values_, not by what they happen to represent when interpreted as a float. could we have a function that checks if the value is set?

Reply:
This check is mainly used to update eeprom when there were no values saved previously (Happens when a new thermocycler has firmware uploaded for the first time).
We do check if the value is set or not in `set_offset_constant`. If we error while writing a byte to eeprom then an error is raised and then `check_saved_offsets` checks whether the error resulted in a garbage/nan value at the address. It then overwrites the garbage value with default values. We don't do anything further if these default values don't get written though. That's a very rare edge case and it'd probably mean the eeprom's damaged anyway so there's really not much that can be repaired by firmware. The errors should take care of informing the user not to proceed further

Reply:
OK, that sounds good. I'd still love to check byte values rather than float representation but that's ok.